frontend/src/map/GameMap.js

```diff
@@ -46,8 +49,10 @@ class GameMap {
         return this.getTile(x, y, z) != undefined;
     }
 
-    getElevation(x, y) {
-        return this.elevationData.get(cellKey(x, y)) || 0;
+    getElevation(x, y, includeTower=false) {
+        const hasTower = !!this.towers.get(cellKey(x, y));
+
+        return this.elevationData.get(cellKey(x, y)) + (includeTower && hasTower ? towerHeight : 0) || 0;
```

Comment:
The fail-safe `|| 0` still seems to work given that `+` has higher precedence over `||` but it's less clear now why it's there, e.g. to handle the undefined from `Map.get()`. I'd suggest breaking it up into variables to make it clearer.

```suggestion
        const cell = cellKey(x, y);
        const mapHeight = this.elevationData.get(cell) || 0;
        const maybeTowerHeight = (includeTower && this.towers.get(cell)) ? towerHeight : 0;

        return mapHeight + maybeTowerHeight ;
```